frame/multisig/src/lib.rs

```diff
@@ -425,6 +434,17 @@ decl_module! {
 				ensure!(m.when == timepoint, Error::<T>::WrongTimepoint);
 				ensure!(m.approvals.len() < threshold as usize, Error::<T>::NoApprovalsNeeded);
 				if let Err(pos) = m.approvals.binary_search(&who) {
+					if m.approvals.len() + 1 == threshold as usize {
+						if let Some(call) = Self::take_call(&call_hash) {
```

Comment:
This should change the weight calculation of the dispatchable.
Suggestion: Introduce a `weight_of::take_call` function (and corresponding `weight_of::store_call`).